modules/kafka/src/main/scala/com/snowplowanalytics/snowplow/sinks/kafka/KafkaSink.scala

```diff
@@ -46,16 +49,21 @@ object KafkaSink {
     Resource.make(make)(p => Sync[F].blocking(p.close))
   }
 
-  private def impl[F[_]: Sync](config: KafkaSinkConfig, producer: KafkaProducer[String, Array[Byte]]): Sink[F] =
+  private def impl[F[_]: Async](
+    config: KafkaSinkConfig,
+    producer: KafkaProducer[String, Array[Byte]],
+    ec: ExecutionContext
+  ): Sink[F] =
     Sink { batch =>
-      Sync[F].interruptible {
```

Comment:
Is it the case that everything executed via `Async` is automatically interruptible?